sql/core/src/test/scala/org/apache/spark/sql/crossdbms/CrossDbmsQueryTestSuite.scala

```diff
@@ -0,0 +1,350 @@
+ * the bottom of this file.
+ *
+ * You need to have a database server up before running this test.
+ * For example, for postgres:
+ * 1. Install PostgreSQL.
+ *   a. On a mac: `brew install postgresql@13`
+ * 2. After installing PostgreSQL, start the database server, then create a role named pg with
+ *    superuser permissions: `createuser -s pg`` OR `psql> CREATE role pg superuser``
+ *
+ * To run the entire test suite:
+ * {{{
+ *   build/sbt "sql/testOnly org.apache.spark.sql.crossdbms.CrossDbmsQueryTestSuite"
+ * }}}
+ *
+ * To re-generate golden files for entire suite, run:
+ * {{{
+ *   SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly org.apache.spark.sql.crossdbms.CrossDbmsQueryTestSuite"
+ * }}}
+ *
+ * To re-generate golden file for a single test, run:
+ * {{{
+ *   SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly org.apache.spark.sql.crossdbms.CrossDbmsQueryTestSuite -- -z describe.sql"
+ * }}}
+ *
+ * To specify a DBMS to use (the default is postgres):
+ * {{{
+ *   REF_DBMS=mysql SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly org.apache.spark.sql.crossdbms.CrossDbmsQueryTestSuite"
+ * }}}
+ */
+// scalastyle:on line.size.limit
+class CrossDbmsQueryTestSuite extends SQLQueryTestSuite with Logging {
+
+  // Note: the below two functions have to be functions instead of variables because the superclass
+  // runs the test first before the subclass variables can be instantiated.
+  private def crossDbmsToGenerateGoldenFiles: String = {
+    val userInputDbms = System.getenv("REF_DBMS")
+    if (userInputDbms != null && userInputDbms.nonEmpty) {
+      assert(CrossDbmsQueryTestSuite.SUPPORTED_DBMS.contains(userInputDbms),
+        s"$userInputDbms is not currently supported.")
+      userInputDbms
+    } else {
+      CrossDbmsQueryTestSuite.DEFAULT_DBMS
+    }
+  }
+  private def customConnectionUrl: String = System.getenv("REF_DBMS_CONNECTION_URL")
+
+  override protected def runQueries(
+    queries: Seq[String],
+    testCase: TestCase,
+    configSet: Seq[(String, String)]): Unit = {
+    val localSparkSession = spark.newSession()
+
+    var runner: Option[SQLQueryTestRunner] = None
+    val outputs: Seq[QueryTestOutput] = queries.map { sql =>
+      val output = {
+        // Use the runner when generating golden files, and Spark when running the test against
+        // the already generated golden files.
+        if (regenerateGoldenFiles) {
+          if (runner.isEmpty) {
+            val connectionUrl = if (customConnectionUrl != null && customConnectionUrl.nonEmpty) {
+              Some(customConnectionUrl)
+            } else {
+              None
+            }
+            runner = Some(CrossDbmsQueryTestSuite.DBMS_TO_CONNECTION_MAPPING(
+              crossDbmsToGenerateGoldenFiles)(connectionUrl))
+          }
+          val sparkDf = spark.sql(sql)
+          val output = runner.map(_.runQuery(sql)).get
+          // Use Spark analyzed plan to check if the query result is already semantically sorted
```

Comment:
please update the comments to be complete sentences ending with punctuation (same everywhere else in this PR)